components/Buttons/src/ButtonThemer/MDCButtonThemer.h

```diff
@@ -0,0 +1,60 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCButton.h"
+#import "MDCButtonColorThemer.h"
+#import "MDCButtonTypographyThemer.h"
+
+/** Defines a readonly immutable interface for button style data to be applied by a themer. */
+@protocol MDCButtonScheming
+
+/** The color scheme to apply to buttons. */
+@property(nonnull, readonly, nonatomic) id <MDCColorScheming> colorScheme;
+
+/** The typography scheme to apply to buttons. */
+@property(nonnull, readonly, nonatomic) id <MDCTypographyScheming> typographyScheme;
+
+/** The corner radius to be applied to buttons. */
+@property(readonly, nonatomic) CGFloat cornerRadius;
+
+/** The minimum height applied to buttons. */
+@property(readonly, nonatomic) CGFloat minimumHeight;
+
+@end
+
+/** Defines the button style data that will be applied to a button by a themer. */
+@interface MDCButtonScheme : NSObject <MDCButtonScheming>
+
+// Redeclare protocol properties as readwrite
+@property(nonnull, readwrite, nonatomic) MDCSemanticColorScheme *colorScheme;
+@property(nonnull, readwrite, nonatomic) MDCTypographyScheme *typographyScheme;
+@property(readwrite, nonatomic) CGFloat cornerRadius;
+@property(readwrite, nonatomic) CGFloat minimumHeight;
+
+@end
```

Comment:
Might be sensible in a follow-up change to break MDCButtonScheme and the protocol out to a separate header now, and introduce a new header for MDCTextButtonThemer.

Reply:
sgtm

Reply:
https://github.com/material-components/material-components-ios/pull/3385